collect_app/src/test/java/org/odk/collect/android/utilities/DaylightSavingTest.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2017 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.odk.collect.android.utilities;
+
+import android.widget.DatePicker;
+import android.widget.TimePicker;
+
+import org.javarosa.core.model.IFormElement;
+import org.javarosa.core.model.QuestionDef;
+import org.javarosa.core.model.data.IAnswerData;
+import org.javarosa.form.api.FormEntryPrompt;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.internal.util.reflection.Whitebox;
+import org.odk.collect.android.BuildConfig;
+import org.odk.collect.android.widgets.DateTimeWidget;
+import org.odk.collect.android.widgets.DateWidget;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.annotation.Config;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.TimeZone;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.stub;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(constants = BuildConfig.class, sdk = 21, manifest = "src/main/AndroidManifest.xml",
+        packageName = "org.odk.collect")
+// https://github.com/opendatakit/collect/issues/356
+// The purpose of this test is to confirm that the app doesn't crash anymore in that case
```

Comment:
I think it would be helpful to give a little bit more information here. Even just saying something like what you said on the method would be great. "Verify that the datetime widget skips over "daylight savings gaps". This is needed on the day and time of a daylight savings transition because that date/time doesn't exist. The date widget doesn't do this skipping."